### DISCUSSION TITLE
Enhanced Water Usage Tracking in Energy Dashboard

### DISCUSSION THREAD

**Discussion by theres** [home-assistant/frontend#20811] [Dashboard]
Currently, the Energy Dashboard allows basic water consumption tracking. For energy, it supports additional information such as solar energy production from panels, home energy storage, and device-specific consumption tracking.

For water, we could have a similarly comprehensive setup:

- We can have a rainwater tank (analogous to energy storage).
- We can track water usage by specific devices and for garden irrigation purposes.
- In some countries, it is supported to have a sub-meter for irreversibly used water, which exempts it from additional sewage charges.
- It is also possible to monitor the amount of produced wastewater.

**Comment by ChristophCaina** [comment]
exactly, what I am looking for in the future...

**Comment by Benon-user** [comment]
In my house I have also 2 water meters that are nested.
In my country the typical solution from water supply companies is to install additional sub counter for garden which is a downstream meter of the main water counter. Then I don’t pay for sewage disposal for the amount of water I use for garden.
I got integrated both the meters in my HA but now the Total water consumption is over calculated.
It would be nice to at least add upstream device feature for water counters similar to energy meters., then dashboard would not summarize water meters reading like is doing now.
![image](https://github.com/user-attachments/assets/0733f592-69b5-4974-89bb-485c9c4f66d2)

**Comment by GiuseppeP96** [comment]
It would be helpful as well as for gas meters.

**Comment by MindFreeze** [comment]
Water hierarchy https://github.com/home-assistant/core/pull/153910

**Comment by Hedda** [comment]
Also see:

- https://github.com/orgs/home-assistant/discussions/273

and

- https://community.home-assistant.io/t/individual-device-energy-usage-for-water/592555

and 

- https://github.com/home-assistant/frontend/discussions/25762
